### PR TITLE
Add support for RPM repomd checksums

### PR DESCRIPTION
**Pull request by sgallagher**
Fixes https://github.com/fedora-modularity/libmodulemd/issues/55

Creates a new object type, `Modulemd.RpmMapEntry` to store NEVRA data and then associates it to the `Modulemd.ModuleStreamV2` object by digest-type and checksum.

### CODE REVIEW

File: spec.v2.yaml [hunk 1]
```diff
@@ -357,3 +357,24 @@ data:
             - xxx-0:1-1.module_deadbeef.x86_64
             - xxx-0:1-1.module_deadbeef.i686
             - xyz-0:1-1.module_deadbeef.x86_64
+        # The rpm-map exists to link checksums from repomd to specific
+        # artifacts produced by this module. Any item in this list must match
+        # a
```
Comment (ignatenkobrain) [inline spec.v2.yaml]: seems like something is missing here.

File: modulemd/v2/include/modulemd-2.0/modulemd-rpm-map-entry.h [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+ *
+ * Create a new rpm-map entry. This will contain the decomposed NEVRA of an
+ * RPM artifact.
+ *
+ * Returns: (transfer full): A newly-allocated #ModulemdRpmMapEntry with the
+ * provided values.
+ */
+ModulemdRpmMapEntry *
+modulemd_rpm_map_entry_new (const gchar *name,
+                            guint32 epoch,
```
Comment (Conan-Kudo) [inline modulemd/v2/include/modulemd-2.0/modulemd-rpm-map-entry.h]: Why isn't this a const as well?

File: spec.v2.yaml (hunk 1 above)
Comment (sgallagher) [reply spec.v2.yaml]: Hmm, I wonder how that happened...

File: modulemd/v2/include/modulemd-2.0/modulemd-rpm-map-entry.h (hunk 2 above)
Comment (sgallagher) [reply modulemd/v2/include/modulemd-2.0/modulemd-rpm-map-entry.h]: The epoch value? The `const` would be meaningless here as the value is copied, not passed by reference.

**Comment by sgallagher** [comment]
Pushed a fix for the missing half a sentence in `spec.v2.yaml`

File: spec.v2.yaml [hunk 3]
```diff
@@ -357,3 +357,24 @@ data:
...
+        # Optional.
+        rpm-map:
+          # The digest-type of this checksum. Required.
+          sha256:
+            # The checksum of the artifact being sought. Required.
+            ee47083ed80146eb2c84e9a94d0836393912185dcda62b9d93ee0c2ea5dc795b:
+              # The RPM name. Required.
+              name: bar
+              # The RPM epoch. Optional
+              epoch: 0
```
Comment (Conan-Kudo) [inline spec.v2.yaml]: I don't know if we should make this property optional. It's not optional in rpm-md, as null epoch is automatically forced to zero epoch.

Comment (sgallagher) [reply spec.v2.yaml]: I made it optional for reading, but libmodulemd outputs it unconditionally. Is that good enough or do you feel strongly that it should be mandatory in all cases?

Comment (Conan-Kudo) [reply spec.v2.yaml]: I think it should remain mandatory in all cases. Remember, this should be consistent for other YAML-based parsers for modulemd.

**Comment by sgallagher** [comment]
I rebased atop the current master branch and made the epoch field mandatory (and validated that it's present during parsing).

**Comment by sgallagher** [comment]
I added the most recent changes as a fixup commit that I'll squash before merging, but I left it separate for ease of review.

**Pull request merged**